core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -153,6 +154,12 @@ public static DataSource createUserStoreDataSource(RealmConfiguration realmConfi
         if (realmConfig.getUserStoreProperty(JDBCRealmConstants.VALIDATION_QUERY) != null) {
             poolProperties.setValidationQuery(realmConfig.getUserStoreProperty(
                     JDBCRealmConstants.VALIDATION_QUERY));
+            poolProperties.setTestOnBorrow(true);
+        }
+        if (StringUtils.isNotEmpty(realmConfig.getUserStoreProperty(JDBCRealmConstants.VALIDATION_INTERVAL)) &&
+            StringUtils.isNumeric(realmConfig.getUserStoreProperty(JDBCRealmConstants.VALIDATION_INTERVAL)) ) {
```

Comment:
Wouldn't it be better to throw an exception and stop the process if it is not numeric ? Otherwise user will have no clue until he goes back and checks.